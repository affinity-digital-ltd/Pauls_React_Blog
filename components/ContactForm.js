import { Component } from 'react'

class ContactForm extends Component {
  render () {
    return <section className='contact-form'>
      <h3>Contact Form</h3>
      <p>Fill in the contact form and I'll get back to you as soon as possible.</p>
      <fieldset>
        <form>
          <input type='text' placeholder='Enter name here' name='name' className='contact-form__input--short' required />
          <input type='email' placeholder='Enter email here' name='email' className='contact-form__input--short' required />
          <textarea placeholder='Enter message here.' className='contact-form__input--tall' name='body' />
          <input type='submit' value='Send' className='contact-form__btn' />
        </form>
      </fieldset>
    </section>
  }
}

export default ContactForm
