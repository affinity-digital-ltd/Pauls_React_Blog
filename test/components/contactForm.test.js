import ContactForm from '../../components/Footer'

describe('<ContactForm />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <ContactForm/>
    )

    expect(component).toMatchSnapshot()
  })
})