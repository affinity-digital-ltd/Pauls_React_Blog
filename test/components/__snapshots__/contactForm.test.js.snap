// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Contact Form
        </h3>,
        <p>
          Fill in the contact form and I'll get back to you as soon as possible.
        </p>,
        <fieldset>
          <form
            onSubmit={[Function]}
          >
            <input
              className="contact-form__input--short"
              name="name"
              placeholder="Enter name here"
              required={true}
              type="text"
            />
            <input
              className="contact-form__input--short"
              name="email"
              placeholder="Enter email here"
              required={true}
              type="email"
            />
            <textarea
              className="contact-form__input--tall"
              name="body"
              placeholder="Enter message here."
              required={true}
            />
            <input
              className="contact-form__btn"
              type="submit"
              value="Send"
            />
          </form>
        </fieldset>,
      ],
      "className": "contact-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Contact Form",
        },
        "ref": null,
        "rendered": "Contact Form",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fill in the contact form and I'll get back to you as soon as possible.",
        },
        "ref": null,
        "rendered": "Fill in the contact form and I'll get back to you as soon as possible.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
          >
            <input
              className="contact-form__input--short"
              name="name"
              placeholder="Enter name here"
              required={true}
              type="text"
            />
            <input
              className="contact-form__input--short"
              name="email"
              placeholder="Enter email here"
              required={true}
              type="email"
            />
            <textarea
              className="contact-form__input--tall"
              name="body"
              placeholder="Enter message here."
              required={true}
            />
            <input
              className="contact-form__btn"
              type="submit"
              value="Send"
            />
          </form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="contact-form__input--short"
                name="name"
                placeholder="Enter name here"
                required={true}
                type="text"
              />,
              <input
                className="contact-form__input--short"
                name="email"
                placeholder="Enter email here"
                required={true}
                type="email"
              />,
              <textarea
                className="contact-form__input--tall"
                name="body"
                placeholder="Enter message here."
                required={true}
              />,
              <input
                className="contact-form__btn"
                type="submit"
                value="Send"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "contact-form__input--short",
                "name": "name",
                "placeholder": "Enter name here",
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "contact-form__input--short",
                "name": "email",
                "placeholder": "Enter email here",
                "required": true,
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "contact-form__input--tall",
                "name": "body",
                "placeholder": "Enter message here.",
                "required": true,
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "contact-form__btn",
                "type": "submit",
                "value": "Send",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "fieldset",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Contact Form
          </h3>,
          <p>
            Fill in the contact form and I'll get back to you as soon as possible.
          </p>,
          <fieldset>
            <form
              onSubmit={[Function]}
            >
              <input
                className="contact-form__input--short"
                name="name"
                placeholder="Enter name here"
                required={true}
                type="text"
              />
              <input
                className="contact-form__input--short"
                name="email"
                placeholder="Enter email here"
                required={true}
                type="email"
              />
              <textarea
                className="contact-form__input--tall"
                name="body"
                placeholder="Enter message here."
                required={true}
              />
              <input
                className="contact-form__btn"
                type="submit"
                value="Send"
              />
            </form>
          </fieldset>,
        ],
        "className": "contact-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Contact Form",
          },
          "ref": null,
          "rendered": "Contact Form",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fill in the contact form and I'll get back to you as soon as possible.",
          },
          "ref": null,
          "rendered": "Fill in the contact form and I'll get back to you as soon as possible.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <input
                className="contact-form__input--short"
                name="name"
                placeholder="Enter name here"
                required={true}
                type="text"
              />
              <input
                className="contact-form__input--short"
                name="email"
                placeholder="Enter email here"
                required={true}
                type="email"
              />
              <textarea
                className="contact-form__input--tall"
                name="body"
                placeholder="Enter message here."
                required={true}
              />
              <input
                className="contact-form__btn"
                type="submit"
                value="Send"
              />
            </form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="contact-form__input--short"
                  name="name"
                  placeholder="Enter name here"
                  required={true}
                  type="text"
                />,
                <input
                  className="contact-form__input--short"
                  name="email"
                  placeholder="Enter email here"
                  required={true}
                  type="email"
                />,
                <textarea
                  className="contact-form__input--tall"
                  name="body"
                  placeholder="Enter message here."
                  required={true}
                />,
                <input
                  className="contact-form__btn"
                  type="submit"
                  value="Send"
                />,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "contact-form__input--short",
                  "name": "name",
                  "placeholder": "Enter name here",
                  "required": true,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "contact-form__input--short",
                  "name": "email",
                  "placeholder": "Enter email here",
                  "required": true,
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "contact-form__input--tall",
                  "name": "body",
                  "placeholder": "Enter message here.",
                  "required": true,
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "contact-form__btn",
                  "type": "submit",
                  "value": "Send",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          "type": "fieldset",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
